tmt/steps/report/reportportal.py

```diff
@@ -143,8 +153,82 @@ def go(self) -> None:
             message = response.text
         if not response.ok:
             raise tmt.utils.ReportError(
-                f"Received non-ok status code from ReportPortal, response text is: {message}")
+                "Received non-ok status code from ReportPortal, "
+                f"response text is: {message}")
         else:
             self.debug(f"Response code from the server: {response.status_code}")
             self.debug(f"Message from the server: {message}")
             self.info("report", "Successfully uploaded.", "yellow")
+
+        try:
+            _ = json.dumps(meta_data)
+        except Exception as exc:
+            raise tmt.utils.SpecificationError("Invalid syntax of provided metadata") from exc
+
+        if meta_data:
+            uuid = None
+            re_result = re.search("\\w{8}-(\\w{4}-){3}\\w{12}", str(message))
+            if isinstance(re_result, re.Match):
+                uuid = re_result.group()
+
+            if not isinstance(uuid, str):
+                self.debug(f"Could not update metadata in {message}")
+                return
+
+            # Get the launch_id from MQ uuid
+            with tmt.utils.retry_session() as session:
+                if self.get("ca-cert-file"):
+                    session.verify = self.get("ca-cert-file")
+                url = f"{server}/api/v1/{project}/launch/uuid/{uuid}"
+                self.debug(f"Send the report to '{url}'.")
+                response = session.get(
+                    url,
+                    headers={
+                        "Authorization": "bearer " + token,
+                        "accept": "*/*",
+                        },
+                    )
+                # Handle the response
+                try:
+                    message = tmt.utils.yaml_to_dict(response.text).get("message")
+                except (tmt.utils.GeneralError, KeyError):
+                    message = response.text
+                if not response.ok:
+                    raise tmt.utils.ReportError(
+                        "Received non-ok status code from ReportPortal, "
+                        f"response text is: {message}")
+                else:
```

Comment:
No need for `else`: if the condition above is true, function would be left through exception, therefore `else` is implicit, and it can be simplified by indenting the calls below oen level to the left:

```python
if not response.ok:
    raise ...

self.debug(...)
```

Reply:
Ack